ttnn/ttnn/multi_device.py

```diff
@@ -154,6 +154,37 @@ def config(self):
         }
 
 
+class ShardTensor2dMesh(TensorToMesh):
+    def __init__(self, device_mesh, shard_grid, shard_dimensions):
+        super().__init__(device_mesh)
+        self.shard_grid = shard_grid  # defines shape of 2D grid of shards
+        self.shard_dimensions = shard_dimensions  # defines which dimensions to shard
+
+    def map(self, tensor):
+        import torch
+
+        Y, X = self.shard_dimensions
+        # Returns list of tensors to map to row-major ordering of chips in shard grid
+        if self.shard_dimensions[Y] is None:
```

Comment:
@cfjchu I think there's a typo here, shouldn't it be Y instead of self.shard_dimensions[Y]?